### PR TITLE
Support retrieving lower bound from TypeVariable in Collection attribute

### PR DESCRIPTION
**Pull request by amoscatelli**
https://github.com/eclipse-ee4j/yasson/issues/579

The issue is related to TypeVariable having a GenericDeclaration different from the expected inherited one.
In the example described in the issue, the GenericDeclaration is FindRequestDTO, and that doesn't match with FindCriteriaDTO.

This is an issue for https://github.com/eclipse-ee4j/yasson/blob/master/src/main/java/org/eclipse/yasson/internal/VariableTypeInheritanceSearch.java#L98

Tell me if you need a more structured approach.

Thank you

### CODE REVIEW

**Comment by Verdent** [comment]
Hi @amoscatelli ,
I am sorry, I am just completely preoccupied on other project. I will definitely check that out once I have time :-) 

Thank you for your contribution!
I am sorry it took me so long to answer and so you have decided to close the PR.

**Comment by amoscatelli** [comment]
@Verdent I closed the issue because I understood the issue is not about typevariable alone, but is about a typevariable representing a generic with the "extends" keyword.

I need to research more, then I'll update the issue I opened before and if you want I'll repone the pull request.

**Comment by amoscatelli** [comment]
@Verdent hi again
I wrote a test to show you the real issue

The issue is with TypeVariable (and **NOT** WildcardVariable) used to define a Collection of a wrapper generic class attribute.
(I reproduced the issue with both Collection and Map to tell the truth).

To get the TypeVariable (and not the WildcardVariable) I use reflection (see the getDeclaredField in the test).

The same actually works if you use the generic to define a non collection attribute, in other words if you remove Collection from the attribute.

The same works if you don't use a wrapper class too.

In other words, these combinations work : 

private Collection<T> value;
private AnotherGenericTestClass<Integer, T> value;
private T value;

This combination doesn't work : 

private Collection<AnotherGenericTestClass<Integer, T>> value;

By patching your internal TypeUtils I made this work, but I don't know if there is something else behind this issue.

Please let me know.

**Comment by redmitry** [comment]
This patch made my day (after loosing another one).

I have a generic API where actual entries are not specified;
```java
public List<BeaconResultset<T>> getResultSets() {
    return result_sets;
}
```
The server may put any jsonb annotated entries, but serialization on server side fails without the patch.

D.

**Review by redmitry — APPROVED** [review]
The patch definitely fixes the issue.

**Review by Verdent — APPROVED** [review]
